### PR TITLE
Fix kernel graphics driver bug; use i.MX fork for wayland-protocols

### PR DESCRIPTION
**Pull request by thochstein**

### CODE REVIEW

**Comment by thochstein** [comment]
> Is the 'Let's fork it' strategy really the path to the future. I give in on that one.

@otavio feels strongly about using forks.

File: recipes-graphics/wayland/wayland-protocols_1.17.imx.bb [hunk 1]
```diff
@@ -9,15 +9,9 @@ LICENSE = "MIT"
...
-           file://0001-linux-dmabuf-support-passing-buffer-DTRC-meta-to-com.patch \
-           "
-SRC_URI[md5sum] = "55ddd5fdb02b73b9de9559aaec267315"
-SRC_URI[sha256sum] = "df1319cf9705643aea9fd16f9056f4e5b2471bd10c0cc3713d4a4cdc23d6812f"
-S = "${WORKDIR}/${ARCHIVE_NAME}"
+SRC_URI = "git://source.codeaurora.org/external/imx/wayland-protocols-imx.git;protocol=https;branch=wayland-protocols-imx-1.17"
+SRCREV = "50ea8fbdd1b768ccaa33dfee43b0f12245ef09b8"
+S = "${WORKDIR}/git"
 
 UPSTREAM_CHECK_URI = "https://wayland.freedesktop.org/releases.html"
```
[the remark this answers is not in the document]
Comment (thochstein) [reply recipes-graphics/wayland/wayland-protocols_1.17.imx.bb]: Agreed, thanks.

File: recipes-graphics/wayland/wayland-protocols_1.17.imx.bb [hunk 2]
```diff
@@ -9,15 +9,9 @@ LICENSE = "MIT"
 LIC_FILES_CHKSUM = "file://COPYING;md5=c7b12b6702da38ca028ace54aae3d484 \
                     file://stable/presentation-time/presentation-time.xml;endline=26;md5=4646cd7d9edc9fa55db941f2d3a7dc53"
 
-ARCHIVE_NAME = "${BPN}-1.17"
-SRC_URI = "https://wayland.freedesktop.org/releases/${ARCHIVE_NAME}.tar.xz \
-           file://0001-unstable-Add-alpha-compositing-protocol.patch \
-           file://0002-unstable-Add-hdr10-metadata-protocol.patch \
-           file://0001-linux-dmabuf-support-passing-buffer-DTRC-meta-to-com.patch \
```
[the remark this answers is not in the document]
Comment (thochstein) [reply recipes-graphics/wayland/wayland-protocols_1.17.imx.bb]: Yes, thanks.

**Comment by MaxKrummenacher** [comment]
@thochstein 
Thanks Tom, my concerns are resolved.

**Comment by otavio** [comment]
The reason of forking is because it forces then to use the Git and we can, if need, merge new upstream releases on top of their changes.

The best is get their changes upstream and no fork at all. But we need to get the best of the current possible solutions.

Do you have a better idea?

**Comment by MaxKrummenacher** [comment]
With a forked downstream git repo it is hard to see what actually changed.
Also, if the changes are small chances are that they apply unchanged even after a version upgrade.

But anyway, as NXP does not share that view it's probably best to go forward with the recipe using the downstream fork in codeaurora.

**Comment by thochstein** [comment]
Ping.

**Comment by petegriffin** [comment]
@thochstein Does this fix a runtime issue? Or it just replaces local patching on top of known upstream release, with fetching from nxp fork repo on codeaurora?

The reason for asking is I see some corruption currently with Weston desktop on imx8m-evk not rendering properly.

**Comment by thochstein** [comment]
> @thochstein Does this fix a runtime issue? Or it just replaces local patching on top of known upstream release, with fetching from nxp fork repo on codeaurora?

The latter, this just replaces local patching. To be sure, I just now did a comparison using `-c patch`.

**Pull request merged**